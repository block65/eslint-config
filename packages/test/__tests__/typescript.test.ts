import { describe, expect, test } from '@jest/globals';
import { ESLint } from 'eslint';
import * as path from 'path';
import { createFixtureLinter } from './helpers.js';

describe('Typescript', () => {
  const fixtureDir = path.resolve(__dirname, './fixtures/typescript');

  const eslint = new ESLint({
    useEslintrc: true,
    cwd: fixtureDir,
    cache: false,
  });

  const fixtureLinter = createFixtureLinter(eslint, fixtureDir);

  test('random TS', async () => {
    const result = await fixtureLinter('random.ts');
    expect(result.messages).toMatchSnapshot();
  });
  test('imports', async () => {
    const result = await fixtureLinter('imports.ts');
    expect(result.messages).toMatchSnapshot();
  });
});
