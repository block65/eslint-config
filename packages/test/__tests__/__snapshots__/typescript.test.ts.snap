// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typescript imports 1`] = `[]`;

exports[`Typescript random TS 1`] = `[]`;
